[Seah Zhi Xuan] iP

joszx opened the pull request:
## MyBoss
> “From being nothing to something, I have travelled a long journey under the guidance of my boss.” - Someone ([source](https://www.europeanbusinessreview.com/inspirational-quotes-for-boss-to-uplift-your-business/))

_**MyBoss** acts as your boss, reminding you of the your list of unending tasks._

**MyBoss** is
- ~~Your Boss~~ your task manager
- text-based
- easy to use

All you need to do is,
1. download it from [here](https://github.com/joszx/ip/releases/download/v0.1/MyBoss.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks like your boss! 😎 

**Features:**
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

#
If you're a Java programmer, you can use it to practice Java too. Here's the main method located in the [MyBoss class](https://github.com/joszx/ip/blob/master/src/main/java/myboss/MyBoss.java).
```java
public static void main(String[] args){
        new MyBoss("./data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

joszx committed:
Duke Level-1

joszx committed:
Fix Level-1, Simplify output

joszx committed:
Duke Level-2

joszx committed:
Added Task class

joszx committed:
Added mark and unmark commands

joszx committed:
Added ToDo class

joszx committed:
Added Deadline

joszx committed:
Added Event class

joszx committed:
Added automated text-ui testing

joszx committed:
Added Exceptions

joszx committed:
Modified Exceptions and fixed Automated Testing

joszx committed:
Added delete and modified output

joszx committed:
Change to switch statement

joszx committed:
Add saving tasks to hard disk feature

joszx committed:
Add loading tasks from hard disk on startup

joszx committed:
Add date handling for event and deadline

joszx committed:
Merge branch 'branch-Level-7'

joszx committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

joszx committed:
Modify OOP structure

joszx committed:
Divide classes into packages

joszx committed:
Add JUnit testing

joszx committed:
Package the App as a JAR file

joszx committed:
Add find command

joszx committed:
Tweak the code to comply with the coding standard

joszx committed:
Add JavaDoc comments

joszx committed:
Merge branch 'branch-A-JavaDoc'

joszx committed:
Merge branch 'branch-A-CodingStandard'

joszx committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/myboss/Ui.java

kaiyichen commented on src/main/java/myboss/Deadline.java:
```diff
@@ -0,0 +1,79 @@
+ * Represents a task with a specified date as deadline. A deadline Object corresponds
+ * to a task with a specified date.
+ */
+public class Deadline extends Task {
```
Should the class name be written in PascalCase? e.g. DeadLine :thinking:

kaiyichen commented on src/main/java/myboss/Deadline.java:
```diff
@@ -0,0 +1,79 @@
+ * to a task with a specified date.
+ */
+public class Deadline extends Task {
+    LocalDate deadline;
```
Should the variable name be in camelCase? e.g. deadLine :no_mouth:

kaiyichen commented on src/main/java/myboss/Event.java:
```diff
@@ -0,0 +1,98 @@
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
```
Perhaps it's better to put on a separate line? e.g.
if (this == o) {
    return true;
}

kaiyichen commented on src/main/java/myboss/Storage.java:
```diff
@@ -0,0 +1,147 @@
+     *
+     * @return whether the clear was successful.
+     */
+    public boolean clearTaskFile() {
```
Perhaps it is better to name a boolean method to sound like boolean? e.g. hasClearTaskFile :relaxed:

kaiyichen commented on src/main/java/myboss/Task.java:
```diff
@@ -0,0 +1,74 @@
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Should this be put as a separate line?

kaiyichen commented on src/main/java/myboss/Task.java:
```diff
@@ -0,0 +1,74 @@
+            this.isDone = true;
+            return  " Nice! I've marked this task as done:" + "\n" + "  " +
+                    "     [" + (this.isDone ? "X" : " ") + "] " +
+                    this.taskName;
```
I like how you break the line !! :thumbsup:

kaiyichen reviewed (commented):
Overall, your code is neat and easy to understand!! Just some nits to fix

ngjunkang commented on src/main/java/myboss/MyBoss.java:
```diff
@@ -0,0 +1,105 @@
+
+        while (true) {
+            String userCmd = ui.getUserCmd();
+            String[] splitUserCmd = Parser.splitUserCmd(userCmd);
```
Verb should not be used in variable names, what do you think about `userCmdSplit` instead?

ngjunkang commented on src/main/java/myboss/MyBoss.java:
```diff
@@ -0,0 +1,105 @@
+                        String timeBy = parsedDeadline[1];
+                        Deadline newDl = new Deadline(deadlineName, timeBy);
+                        tasks.addTask(newDl);
+                        ui.addTaskOutput(newDl);
```
I feel that the `l` in `newDl` can be ambiguous with `1`, perhaps you can spell it out like the others, `newDeadline`?

ngjunkang commented on src/main/java/myboss/Ui.java:
```diff
@@ -0,0 +1,150 @@
+     *
+     * @param output string to be outputted.
+     */
+    static void myBossOutput(String output) {
```
Verbs should be used for method names, perhaps you can consider `outputMyBoss` (consistent with other methods in Ui)? With that said, I don't really like the verb `output`, maybe `print` is a better word, hence `printCmdOutput` may be more meaningful :)

ngjunkang commented on src/main/java/myboss/Ui.java:
```diff
@@ -0,0 +1,150 @@
+    /**
+     * Outputs a formatted line break.
+     */
+    static void hLineBreak() {
```
What is `h` in `hLineBreak`?

ngjunkang commented on src/main/java/myboss/Ui.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    /**
+     * Output the given task as unmarked.
```
Does `Output` have another form `Outputs`? If it doesn't, the Javadoc for `hLineBreak` method and some other places are inconsistent.

ngjunkang commented on test/myboss/ParserTest.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @Test
+    void testGetRemainingUserCmd_ExitCase_EmptyString() {
```
Do you know that there's a format to follow for test case method name as stated in Week 3 readings? E.g. `whatIsBeingTested_descriptionOfTestInputs_expectedOutcome`. I will probably name this as `getRemainingUserCmd_exitCommand_emptyString` and do the same for the other test cases.

ngjunkang commented on src/main/java/myboss/MyBoss.java:
```diff
@@ -0,0 +1,105 @@
+            String remainingUserInput = Parser.getRemainingUserCmd(splitUserCmd);
+
+            try {
+                switch(command) {
+                    case "bye":
+                        storage.updateFile(tasks.getTaskList());
+                        ui.outputFarewell();
+                        System.exit(0);
+                    case "list":
+                        ui.outputTaskList(tasks.getTaskList());
+                        break;
```
Perhaps you can make this switch statement compliant to the Java coding standard (no case indentations)? e.g.
```java
switch (type) {
case "bye":
    break;
case "list":
    break;
...
}
```

ngjunkang reviewed (commented):
LGTM, just some nits to fix according to the desired coding quality, good job! 👍

joszx committed:
Fix some nits according to code review

joszx committed:
Add gradle wrapper file

joszx committed:
Update runtest of text-ui-test

joszx committed:
Merge branch 'add-gradle-support'

# Conflicts:
#	.gitignore

joszx committed:
Automate project builds and unit tests using Gradle

joszx committed:
Merge branch 'A-Gradle'

joszx committed:
Add simple GUI with JavaFX

joszx committed:
Modify GUI to use FXML

joszx committed:
Add MyBoss features to GUI

joszx committed:
Fix gradle to run from launcher

joszx committed:
Add assertion

There is no use of assertions.

Assertions will document important assumptions that should hold at various points in the code.

As a step towards improving code quality, lets add the use of assertion in Parser class.

joszx committed:
Improve Code Quality

There were many areas in the code that can be modified to improve code
quality and readability.

Improving code quality is more developer friendly because future developers
will have to read and understand the code for future development.

Let's,
* remove deep nesting
* remove magic literal strings
* make happy path prominent by adding guard clauses
* fix empty catch blocks
* delete dead code
* minimize code duplication

joszx committed:
Add the use of Java Streams

There is a method that can be modified to use Java Streams.

Using Java Streams will make the code more concise whilst allowing for
easier modifications in the future.

Modified the findTasks method in TaskList class to use Java Streams.

joszx committed:
Merge pull request #2 from joszx/branch-A-Assertions

Add assertion

joszx committed:
Merge branch 'master' of https://github.com/joszx/ip

joszx committed:
Merge pull request #3 from joszx/branch-A-CodeQuality

Improve Code Quality

joszx committed:
Merge branch 'master' of https://github.com/joszx/ip

joszx committed:
Merge pull request #4 from joszx/branch-A-Streams

Add the use of Java Streams

joszx committed:
Create gradle.yml

joszx committed:
Update gradle.yml

joszx committed:
Update gradle.yml

joszx committed:
Add checkstyle to gradle and fixed checkstyle violations

joszx committed:
Merge branch 'branch-A-CheckStyle'

joszx committed:
Merge branch 'master' of https://github.com/joszx/ip

joszx committed:
Update gradle.yml

joszx committed:
Fix storage test case

joszx committed:
Change JavaFX label to text to allow longer messages

joszx committed:
Add priority command

joszx committed:
Merge branch 'branch-C-Priority'

joszx committed:
Improve the GUI

joszx committed:
Fix minor coding style violations

joszx committed:
Add user guide
